app/main.py

```diff
@@ -2,24 +2,67 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int,
+                 max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type[object],
+                     name: str) -> None:
+        self.protect_name = "_" + name
+
+    def __get__(self, instance: object,
+                owner: type[object]) -> int:
+        return getattr(instance, self.protect_name)
+
+    def __set__(self, instance: object,
+                value: int) -> None:
+        if self.min_amount <= value <= self.max_amount:
+            setattr(instance, self.protect_name, True)
+        else:
+            setattr(instance, self.protect_name, False)
```

Comment:
The IntegerRange class should store the value assigned to it, rather than just a boolean indicating whether the value is within the range. Besides that, the descriptor should raise appropriate exceptions when attempting to assign invalid values. Reformat your code to encounter that

Reply:
In order to process ValueError, try-except should be used. But exception handling in the next topic, so i tried to solve this problem in different way :)